### DISCUSSION TITLE
running the cypress via node in C# giving error "Could not find Cypress test run results"

### DISCUSSION THREAD

**Discussion by ArunVenkat1994** [cypress-io/cypress#26648] [Questions and Help]
Hi, 

I am trying to run the cypress from C# using the node package "Jering.Javascript.NodeJS" 

When I run the cypress from cmd directly, it is running fine and giving results. 

but when I run from c#, I am getting the below error. 
{"status":"failed","failures":1,"message":"Could not find Cypress test run results"}

**C# code:** 
using Jering.Javascript.NodeJS;

string result = await StaticNodeJSService.InvokeFromFileAsync<string>(@"E:\Samples\Cypress\E2E\e2e-run-tests.js");

**JS code: 
E:\Samples\Cypress\E2E\e2e-run-tests.js file code:**

const cypress = require('cypress')
module.exports = async function(callback){ 
  let result = ""; 

  await cypress
   .run({
     spec: 'Test1.cy.js',
   })
    .then(testResults => {
     result = testResults;
   })
    .catch((err) => {
     callback(err); 
   })
   return result;
}

**Test1.cy.js file code:** 
describe('empty spec', () => {
  it('passes', () => {
    cy.visit('https://example.cypress.io')
    cy.contains('type').click()
  })
})

Please suggest the possible solution

**Comment by portyu9** [comment]
Hi @ArunVenkat1994 

The returned object:

```json
{
  "status": "failed",
  "failures": 1,
  "message": "Could not find Cypress test run results"
}
```

is a Cypress **failed-run result**, so the C# → Node invocation is getting far enough to call `cypress.run()`.

The first thing I would fix is the working-directory/project-path ambiguity.

With the Cypress Module API:

```js
cypress.run({
  spec: 'Test1.cy.js'
})
```

the `spec` path is resolved relative to the Node process's **current working directory**. When Node is started through Jering from C#, that is not necessarily the same directory you use when running Cypress manually.

Make both paths explicit:

```js
const path = require('path')
const cypress = require('cypress')

module.exports = async () => {
  // Adjust this if e2e-run-tests.js is not in the Cypress project root
  const projectRoot = path.resolve(__dirname)

  const results = await cypress.run({
    project: projectRoot,
    spec: path.join(
      projectRoot,
      'cypress',
      'e2e',
      'Test1.cy.js'
    ),
  })

  if (results.status === 'failed') {
    throw new Error(results.message)
  }

  return JSON.stringify(results)
}
```

Also don't mix Jering's two invocation styles.

Your code currently exports:

```js
module.exports = async function(callback) {
```

Jering supports either:

```js
module.exports = (callback) => {
  callback(null, result)
}
```

**or** an async function:

```js
module.exports = async () => {
  return result
}
```

For `cypress.run()`, the async form is cleaner. If it throws, let the exception propagate rather than calling a callback from an `async` function.

On the C# side you can alternatively deserialize the returned Cypress result into a DTO instead of converting the entire object to a string.

I would also temporarily log:

```js
console.log('cwd:', process.cwd())
console.log('project:', projectRoot)
```

That will quickly confirm whether invoking through the .NET process changed the path Cypress is resolving.

So the strongest diagnosis from the code shown is **project/spec resolution from the Jering Node process**, not a failure of Cypress to execute through Node generally.

There were no screenshots or follow-up comments containing another confirmed cause.

Please mark this answer as accepted if it helped, thank you.
